can-view-parser.js

```diff
@@ -307,6 +307,21 @@ var camelCaseToSpinalCase = function (match, lowerCaseChar, upperCaseChar) {
 	return lowerCaseChar + "-" + upperCaseChar.toLowerCase();
 };
 
+HTMLParser.sanitizeAttrs = function(html) {
+	// represent:
+	// <tag attrs="special characters that can muck up parsing"></tag>
+	// as:
+	// <tag attrs=.............................................></tag>
+	//
+	// regex explanation:
+	// (['"]) capture an opening ' or "
+	// (?:(?!\1).)* match zero or more characters that aren't the captured ' or "
+	// \1 match the closing ' or "
+	return html.replace(/(['"])(?:(?!\1).)*\1/g, function(match) {
```

Comment:
I'm not sure if newline is needed but this allows newlines and allows escaped quotes inside the attr value:
`/(['"])(?:(?:\\[\s\S])|[^\\])*?\1/g`

```
	// regex explanation:
	// (['"]) - capture an opening ' or "
	// (?:(?:\\[\s\S])|[^\\]) - escape slash followed by anything OR a single character that's not an escape slash
	// *? - match the previous thing only as many times as needed to satisfy the full regex
	// \1 - backreference of the captured quote type is last character needed for a match to happen
	//
	// matches:
	//   quoteChar + ( any 2 characters if the first one is '\' OR any character that's not '\' )*? + same quoteChar
```

I don't know what html would look like at this point but if it could be this, it would break both our regexes:
`<span>'</span><span attr='>'></span>`

If it needs to handle that (if html is more similar to a full document rather than a single chunk), I'd be happy to help with that too!